core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -199,8 +199,8 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     numExecutors = Option(numExecutors)
       .getOrElse(sparkProperties.get("spark.executor.instances").orNull)
     queue = Option(queue).orElse(sparkProperties.get("spark.yarn.queue")).orNull
-    keytab = Option(keytab).orElse(sparkProperties.get("spark.yarn.keytab")).orNull
-    principal = Option(principal).orElse(sparkProperties.get("spark.yarn.principal")).orNull
+    keytab = Option(keytab).orElse(sparkProperties.get("spark.kerberos.keytab")).orNull
```

Comment:
This has a similar issue as the R code above re: backwards compatibility.

Reply:
This might be easier to fix though; `prepareSubmitEnvironment` (where these things are used) could just read them from `SparkConf` at that point. Something like this in that method, instead of this code:

    Option(args.keytab).orElse(sparkConf.get(KEYTAB))

Reply:
agreed, shouldn't the "old" config still work? `spark.yarn.keytab` etc